GitExtUtils/DebugHelpers.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System.Diagnostics;
+
+namespace GitExtUtils;
+
+/// <summary>
+///  Set of DEBUG-only helpers.
+/// </summary>
+public static class DebugHelpers
+{
+    [Conditional("DEBUG")]
+    public static void Assert(bool condition, string message)
+    {
+        if (Debugger.IsAttached)
+        {
+            Debug.Assert(condition, message);
+        }
+        else if (!condition)
+        {
+            Debugger.Launch();
+
+            if (!Debugger.IsAttached)
+            {
+                throw new InvalidOperationException(message);
+            }
+        }
+    }
+
+    [Conditional("DEBUG")]
+    public static void Fail(string message)
+    {
+        if (Debugger.IsAttached)
+        {
+            Debug.Fail(message);
+        }
+        else
+        {
+            Debugger.Launch();
+
+            if (!Debugger.IsAttached)
+            {
+                throw new InvalidOperationException(message);
+            }
```

Comment:
This will prompt to attach a debugger, and if that's declined an exception will get routed to NBug... HOWEVER, if the code that triggered this was executing on a background thread it'd lead to the termination of the app.

Reply:
> HOWEVER, if the code that triggered this was executing on a background thread it'd lead to the termination of the app.

It terminates the app but *after* starting the BugReporter, isn't it?

The goal is to not have background threads without exception handling. I assume that there is (almost) no such background operation left since the recent PR for `JTF.RunAsync` and `FileAndForget`.

Reply:
> It terminates the app but _after_ starting the BugReporter, isn't it?

Correct.